### DISCUSSION TITLE
Uninstall rich.traceback?

### DISCUSSION THREAD

**Discussion by LarsHill** [Textualize/rich#1947] [Q&A] [answered]
Is there a way to return to standard tracebacks (without `rich`) after executing the following code once?
```python
from rich.traceback import install
install()
```

I tried commenting out the above block and re-run the code, but tracebacks are still rendered via `rich`. So it seems `install()` is rather permanant. I also looked at the code of `install()` and saw it overwrites `sys.excepthook`. Hence, I again commented out the above code block and in the `main()` I reset `sys.excepthook` to its default implementation:

```python
def excepthook(exctype, value, traceback): # real signature unknown; restored from __doc__
    """
    excepthook(exctype, value, traceback) -> None
    
    Handle an exception by displaying it with a traceback on sys.stderr.
    """
    pass
```

Of course I could recreate the environment and this would probably work, but I would like to give the user the option to flexibly  toggle on/off the use of `rich` tracebacks in-between runs. Thus, I am looking for a convenient "toggle off" method. E.g. sth like

```python
from rich.traceback import uninstall
uninstall()
```
which could be called conditionally on some user input.

Or maybe there is an other workaround?

**Comment by willmcgugan** [comment, marked as answer]
IIRC install returns the original excepthook, so you can restore it afterwards.

However, you may be better off handling exceptions explicitly in your code. Assuming you have some top level entry point.

```python
try:
    run()
except Exception:
    if rich_exceptions:
        console.print_exception()
    else:
        raise
```

**Reply by LarsHill** [reply]
Thanks, I'll give that second option a try. 

Just to clarify: I don't use `from rich.traceback import install; install()` anymore to overwrite traceback behaviour globally but catch exceptions at the highest level and either raise them via rich `console.print_exception()` or normally `raise`. 
Did I understand you correctly?

> IIRC install returns the original excepthook, so you can restore it afterwards.

This line I don't understand. Since `install()` was executed before in the environment and I don't want to create it from scratch, how can I undo it? How do I envoke "IIRC install" to get the original excepthook? Do you maybe have a code example to do that?

**Reply by LarsHill** [reply]
Ah, nevermind, I noticed I called `from rich.traceback import install; install()` in more than one module. When commenting it out everywhere, Tracebacks are printed normally again. 
I was under the wrong impression that somehow `install()` changes the Traceback behaviour in the environment and not in the concrete python session, which of course does not really make lot's of sense.

**Comment by pstanisl** [comment]
What should I do if I run a package (in this case, Kedro) installing `rich.traceback` in a Databricks Notebook? The issue in the code does not cause notebook cell execution error, i.e., following cells run because everything seems okay. An automation run of this error handling is almost impossible. 

Is it possible to revert the installation of `rich.traceback`?

**Reply by antonymilne** [reply]
@pstanisl I've made a new issue for this here: https://github.com/Textualize/rich/issues/2461. Please do add any comments you might have 🙂
